modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/milker.dm

```diff
@@ -0,0 +1,300 @@
+/obj/item/milker/proc/bottle_check(mob/living/carbon/being_milked, mob/living/carbon/human/milker, obj/item/reagent_containers/milk_bottle)
+	if(!istype(milk_bottle)) // Not a container?
+		return FALSE
+	if(!milk_bottle.is_open_container()) // Sealed container?
+		to_chat(milker, span_warning("[milk_bottle] is sealed!"))
+		return FALSE
+	if(milk_bottle.reagents?.holder_full()) // Is full?
+		to_chat(milker, span_warning("[milk_bottle] is full!"))
+		return FALSE
+	return TRUE
+
+/// Attempt to consume the contents of someone's breasts.
+/obj/item/milker/proc/drink_milk(mob/living/carbon/human/being_milked, mob/living/carbon/human/milker, obj/item/organ/genital/breasts/breasts)
+	if(!breast_check(being_milked, milker, breasts) || !mouth_check(milker))
+		return FALSE
+
+	var/self_suckle = (being_milked == milker) // Feeding off your own supply?
+
+	switch(handle_mode)
+		if(HANDLE_LIGHT) // Gentle paws...
+			if(milker.combat_mode)
+				milker.visible_message(
+				message = span_purple("[milker] yanks one of [self_suckle ? "[milker.p_their()] own" : "[being_milked]'s"] nipples up to [milker.p_their()] mouth and starts nibbling..."),
+				self_message = span_purple("You yank one of [self_suckle ? "your own" : "[being_milked]'s"] nipples up to your mouth and start nibbling..."),
+				blind_message = span_purple("You hear an awkward nibbling noise."),
+				vision_distance = 1) // Subtle distance
+			else
+				milker.visible_message(
+				message = span_purple("[milker] pulls one of [self_suckle ? "[milker.p_their()] own" : "[being_milked]'s"] nipples up to [milker.p_their()] lips..."),
+				self_message = span_purple("You pull one of [self_suckle ? "your own" : "[being_milked]'s"] nipples up to your lips and start suckling..."),
+				blind_message = span_purple("You hear an awkward kissing noise."),
+				vision_distance = 1)
+		if(HANDLE_STRONG) // SQUISH
+			if(milker.combat_mode) // CHOMP
+				milker.visible_message(
+				message = span_purple("[milker] grips hard into one of [self_suckle ? "[milker.p_their()] own" : "[being_milked]'s"] breasts and wrenches it up to [milker.p_their()] mouth, chomping down hard!"),
+				self_message = span_purple("You wrench one of [self_suckle ? "your own" : "[being_milked]'s"] breasts up to your mouth and start to bite down!"),
+				blind_message = span_purple("You hear an awkward chomping noise."),
+				vision_distance = 1) // Subtle distance
+			else // SUCK
+				milker.visible_message(
+				message = span_purple("[milker] clasps a hand into one of [self_suckle ? "[milker.p_their()] own" : "[being_milked]'s"] breasts, then plunges [milker.p_their()] mouth around [milker.p_their()][self_suckle ? " own" : ""] nipple!"),
+				self_message = span_purple("You grab onto one of [self_suckle ? "your own" : "[being_milked]'s"] breasts and ram that nipple right into your mouth!"),
+				blind_message = span_purple("You hear an awkward plapping noise."),
+				vision_distance = 1)
```

Comment:
I also feel like this is a loooot of detail that can be condensed. Maybe a list of actions `("clasps a hand into", "grips hard into", "pulls")` that can be chosen depending on the handle mode?

Reply:
I'm not sure how to do that. Best I can think of is a huge nested list and a bunch of conditionals pulling from that list. How did you mean I do this?

Reply:
I did mean that but thinking a bit more about it I'm not sure it'd be any better